spyder/plugins/editor/widgets/completion.py

```diff
@@ -0,0 +1,407 @@
+                     CompletionItemKind.VARIABLE: 'variable',
+                     CompletionItemKind.CLASS: 'class',
+                     CompletionItemKind.INTERFACE: 'interface',
+                     CompletionItemKind.MODULE: 'module',
+                     CompletionItemKind.PROPERTY: 'property',
+                     CompletionItemKind.UNIT: 'unit',
+                     CompletionItemKind.VALUE: 'value',
+                     CompletionItemKind.ENUM: 'enum',
+                     CompletionItemKind.KEYWORD: 'keyword',
+                     CompletionItemKind.SNIPPET: 'snippet',
+                     CompletionItemKind.COLOR: 'color',
+                     CompletionItemKind.FILE: 'filenew',
+                     CompletionItemKind.REFERENCE: 'reference',
+                     }
+
+        height = self.item_height
+        width = self.item_width
+
+        for completion in self.completion_list:
+            if not self.is_internal_console:
+                completion_label = completion['filterText']
+                icon = icons_map.get(completion['kind'], 'no_match')
+                completion_data = completion['insertText']
+                completion_text = self.get_html_item_representation(
+                    completion_data, icon, height=height, width=width)
+                item = QListWidgetItem(ima.icon(icon),
+                                       completion_text)
+                item.setData(Qt.UserRole, completion_data)
+            else:
+                completion_label = completion[0]
+                completion_text = self.get_html_item_representation(
+                    completion_label, '', height=height, width=width)
+                item = QListWidgetItem()
+                item.setData(Qt.UserRole, completion_label)
+                item.setText(completion_text)
+
+            if self.check_can_complete(
+                    completion_label, filter_text):
+                self.addItem(item)
+
+        if self.count() > 0:
+            self.setCurrentRow(0)
+            self.scrollTo(self.currentIndex(),
+                          QAbstractItemView.PositionAtTop)
+        else:
+            self.hide()
+
+    def get_html_item_representation(self, item_completion, item_type,
+                                     height=DEFAULT_COMPLETION_ITEM_HEIGHT,
+                                     width=DEFAULT_COMPLETION_ITEM_WIDTH):
+        """Get HTML representation of and item."""
+
+        return COMPLETION_ITEM_TEMPLATE.format(completion=item_completion,
+                                               type=item_type,
+                                               color=ima.MAIN_FG_COLOR,
+                                               height=height,
+                                               width=width)
+
+    def hide(self):
+        """Hide the widget."""
+        self.completion_position = None
+        self.completion_list = None
+        self.clear()
+        self.textedit.setFocus()
```

Comment:
```suggestion
        self.textedit.setFocus()
        tooltip = getattr(self.textedit, 'tooltip_widget', None)
        if tooltip:
            tooltip.hide()

```

Reply:
@dalthviz this is so that when pressing Escape, the tooltip also hide. (which was not the case before)